src/seedu/addressbook/ui/TextUi.java

```diff
@@ -142,11 +140,18 @@ private void showPersonListView(List<? extends ReadOnlyPerson> persons) {
 
     /** Shows a list of strings to the user, formatted as an indexed list. */
     private void showToUserAsIndexedList(List<String> list) {
-        showToUser(getIndexedListForViewing(list));
+        showToUser(formatter.getIndexedListForViewing(list));
     }
 
     /** Formats a list of strings as a viewable indexed list. */
-    private static String getIndexedListForViewing(List<String> listItems) {
+    formatter newFormatter = new formatter(listItems);
+}
+class formatter {
```

Comment:
New classes should be added into a new file, unless it is a nested class. If not the code becomes very hard to maintain.